blackbird_python/blackbird/listener.py

```diff
@@ -273,18 +273,22 @@ def exitExpressionvar(self, ctx: blackbirdParser.ExpressionvarContext):
         elif ctx.nonnumeric():
             value = _literal(ctx.nonnumeric())
 
-        try:
-            # assume all variables are scalar
-            final_value = PYTHON_TYPES[vartype](value)
-        except TypeError:
+        if isinstance(value, sym.Expr):
+            # If the expression is a parameter, save it in the variables list _VAR
+            final_value = value
+        else:
             try:
-                # maybe one of the variables was a NumPy array?
-                final_value = NUMPY_TYPES[vartype](value)
-            except:
-                # nope
-                raise TypeError(
-                    "Var {} = {} is not of declared type {}".format(name, value, vartype)
-                ) from None
+                # assume all variables are scalar
+                final_value = PYTHON_TYPES[vartype](value)
+            except TypeError:
+                try:
+                    # maybe one of the variables was a NumPy array?
+                    final_value = NUMPY_TYPES[vartype](value)
+                except:
+                    # nope
```

Comment:
lol. This comment basically amounts to, 'something went wrong, but I do not know what'. Definitely something we can improve on in blackbird!